### DISCUSSION TITLE
Multi-AZ standalone setup

### DISCUSSION THREAD

**Discussion by tuxillo** [loxilb-io/loxilb#872] [General]
Hi,

I was checking the multi-AZ document (https://github.com/loxilb-io/loxilbdocs/blob/main/docs/aws-multi-az.md) and I was wondering if one could setup a multi-AZ but with 3-AZs instead of two. It would also be a standalone installation instead since there would be no kubernetes in this case.

Thanks!

**Comment by UltraInstinct14** [comment]
Currently in stand-alone mode, only up-to two HA instances are supported.  With kube-loxilb operator (in Kubernetes), there is no limit. Given the current limitation, one over-the-top possibility could be running Kubernetes (not EKS but some light weight distro) in all the  loxilb EC2 nodes just for the sake of running kube-loxilb and potentially achieve what you are suggesting. Of course, it would add the overhead of managing Kubernetes on the loxilb nodes itself.

**Reply by tuxillo** [reply]
Not sure I get what you mean, you mean using the kube-loxilb with in-cluster mode?

I've been having a couple other thoughts about how to approach the architecture because there are additional factors such as cross-AZ traffic which has a cost.

One idea would be to have one external active LoxiLB per AZ, each one having their backing cross-zone, that way if an AZ fails, the backups is in a different AZ which (hopefully) would not be affected. But then I would have to have one service per AZ, although sounds suboptimal to have this approach.

**Reply by backguynn** [reply]
I will elaborate on the approach @UltraInstinct14 mentioned.   

Deploy Kubernetes on nodes located in three different Availability Zones (AZs). Then, deploy LoxiLB in in-cluster mode (adding the --cloud=aws --cloudcidrblock=124.124.124.0/24 options to the manifest file), and deploy kube-loxilb alongside it. In this setup, kube-loxilb manages HA, making HA across the three AZs possible.

However, the --cloud=aws option for LoxiLB has not been tested in in-cluster mode. This presents a potential risk.
